src/box/lua/iproto.lua

```diff
@@ -0,0 +1,149 @@
+-- Namespace for working with Tarantool network subsystem.
+
+box.iproto = {
+    -- `IPROTO_FLAGS` bitfield constants enumeration.
+    FLAG_COMMIT            = 0x01,
+    FLAG_WAIT_SYNC         = 0x02,
+    FLAG_WAIT_ACK          = 0x04,
```

Comment:
> I think it's better to add a new namespace for these fields `box.iproto.flag`, like for features, instead of using the same prefix `FLAG_` for each of them

Please ask @unera's opinion first.

Reply:
My 2C: perhaps, we should add subnamespaces for each enumeration?

Reply:
sepatate constants between namespaces - was my first proposal

But You (@locker) and @alyapunov refused it and decided to have a flat dictionary of constants.

So I don't what can I comment here :)

Hmm. Ok. Let's use namespaces! Heyhoo!!

Reply:
In C, we don't have namespaces so we use a prefix instead, e.g. `IPROTO_FEATURE_` or `IPROTO_FLAG_` or `IPROTO_`. IMO it'd be logical to put all constants that have the same prefix in C in the same Lua namespace, for example:

```
IPROTO_OK -> box.iproto.OK
IPROTO_DATA -> box.iproto.DATA
IPROTO_FLAG_COMMIT -> box.iproto.flag.COMMIT
IPROTO_BALLOT_IS_RO -> box.iproto.ballot.IS_RO
IPROTO_FEATURE_STREAMS -> box.iproto.feature.streams
```

(For features we decided to use lowercase, because we already do in net.box)

Reply:
Added subnamespaces for the following enumerations: `IPROTO_FLAGS`, `iproto_metadata_key`, `iproto_ballot_key`, `iproto_raft_keys`.

Reply:
Discussed with @CuriousGeorgiy. Agreed that it's best to separate all C enum constants by namespaces to avoid confusion. Here's how namespaces are going to be named:

* `enum iproto_key` => `box.iproto.key`
  Examples: `box.iproto.key.REQUEST_TYPE, box.iproto.key.SYNC, box.iproto.key.TERM`
* `enum iproto_type` => `box.iproto.type`
  Examples: `box.iproto.type.SELECT, box.iproto.type.RAFT_CONFIRM, box.iproto.type.ID`
* `enum iproto_metadata_key` => `box.iproto.metadata_key`
  Examples: `box.iproto.metadata_key.NAME, box.iproto.metadata_key.IS_NULLABLE`
  *!! Decided to use `box.iproto.metadata_key` namespace rather than `box.iproto.field`, because these fields are passed in `IPROTO_METADATA` !!*
  **?? Do we need to use prefix `FIELD_` here, like we do in `enum iproto_metadata_key`. Probably not, because they all have the same prefix so it bears no additional meaning ??**
* `enum iproto_ballot_key` => `box.iproto.ballot_key`
  Examples: `box.iproto.ballot_key.IS_RO_CFG, box.iproto.ballot_key.CAN_LEAD`
* `enum iproto_raft_keys` => `box.iproto.raft_key`
  Examples: `box.iproto.raft_key.TERM, box.iproto.raft_key.IS_LEADER_SEEN`

Reply:
Added a subnamespace for each enumeration, as discussed.